### PR TITLE
Revert and disable batch enable/disable functionality

### PR DESCRIPTION
**Pull request by qqu0127**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:
Fixes https://github.com/apache/helix/issues/2136 

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

Helix batch instance enable/disable is breaking backward compatibility in release 1.0.3. We'd want to revert and disable this feature for now.

Logic change: 
Instance enable/disable is controlled by instance config only. 
Instance enable/disable is only written to instance config.
Rest call to enable/disable a batch of instances is disabled.

### Tests

- [x] The following tests are written for this issue:

(List the names of added unit/integration tests)

- The following is the result of the "mvn test" command on the appropriate module:

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Helix 1.0.5-SNAPSHOT:
[INFO] 
[INFO] Apache Helix ....................................... SUCCESS [  0.792 s]
[INFO] Apache Helix :: Metrics Common ..................... SUCCESS [  2.656 s]
[INFO] Apache Helix :: Metadata Store Directory Common .... SUCCESS [ 13.464 s]
[INFO] Apache Helix :: ZooKeeper API ...................... SUCCESS [14:20 min]
[INFO] Apache Helix :: Helix Common ....................... SUCCESS [  1.729 s]
[INFO] Apache Helix :: Core ............................... SUCCESS [  01:58 h]
[INFO] Apache Helix :: Admin Webapp ....................... SUCCESS [  1.152 s]
[INFO] Apache Helix :: Restful Interface .................. SUCCESS [05:14 min]
[INFO] Apache Helix :: Distributed Lock ................... SUCCESS [02:23 min]
[INFO] Apache Helix :: HelixAgent ......................... SUCCESS [  0.352 s]
[INFO] Apache Helix :: Recipes ............................ SUCCESS [  0.011 s]
[INFO] Apache Helix :: Recipes :: Rabbitmq Consumer Group . SUCCESS [  1.652 s]
[INFO] Apache Helix :: Recipes :: Rsync Replicated File Store SUCCESS [  1.586 s]
[INFO] Apache Helix :: Recipes :: distributed lock manager  SUCCESS [  1.481 s]
[INFO] Apache Helix :: Recipes :: distributed task execution SUCCESS [  1.417 s]
[INFO] Apache Helix :: Recipes :: service discovery ....... SUCCESS [  1.535 s]
[INFO] Apache Helix :: View Aggregator .................... SUCCESS [02:26 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:23 h
[INFO] Finished at: 2022-06-02T14:23:03-07:00
[INFO] ------------------------------------------------------------------------

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java [hunk 1]
```diff
@@ -120,17 +121,17 @@ public static String getInstanceHelixDisabledType(HelixDataAccessor dataAccessor
    * @param clusterConfig
```
Comment (xyuanlu) [inline helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java:121]: Maybe `getInstanceHelixDisabledType` need change as well. We probably don't want to read from cluster config here?

File: helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java [hunk 2]
```diff
@@ -1,177 +0,0 @@
...
-import org.apache.helix.util.InstanceValidationUtil;
-import org.testng.Assert;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Test;
-
-public class TestBatchEnableInstances extends TaskTestBase {
-  private ConfigAccessor _accessor;
-
-  @BeforeClass
-  public void beforeClass() throws Exception {
```
Comment (xyuanlu) [inline helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java:43]: Let's keep the class but mark tests are disabled.

File: helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java [hunk 3]
```diff
@@ -120,17 +121,17 @@ public static String getInstanceHelixDisabledType(HelixDataAccessor dataAccessor
...
    * @return
    */
-  public static boolean isInstanceEnabled(InstanceConfig instanceConfig, ClusterConfig clusterConfig) {
+  public static boolean isInstanceEnabled(InstanceConfig instanceConfig, @Nullable ClusterConfig clusterConfig) {
     if (instanceConfig == null) {
```
Comment (xyuanlu) [inline helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java:125]: I feel like marking nullable for clusterConfig is a bit conflict with `isEnabled()` line 78. As we are throwing exception if clusterConfig is null. Probably we want to keep the same behavior..?

**Comment by xyuanlu** [comment]
Thanks for making this change! May I suggest let's add comment section to explain the reason why we want to disable this and what are pre-requisite if we want to enable batch in the future?

File: helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java (hunk 3 above)
Comment (qqu0127) [reply helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java:125]: Thanks. The method is called from other places too, so clusterConfig could be null. Let me drop this change, it's not essential anyway.

File: helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java (hunk 1 above)
Comment (qqu0127) [reply helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java:121]: I don't get it. Method `getInstanceHelixDisabledType` is reading clusterConfig for disabled type. How can we not read from cluster config?

File: helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java (hunk 2 above)
Comment (xyuanlu) [reply helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java:43]: Maybe we could try to remove the line added in PR #2006? Can we try to make this change as close to revert commit as possible?

File: helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java (hunk 1 above)
Comment (desaikomal) [reply helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java]: No optimization as this needs to be pure revert.

Comment (xyuanlu) [reply helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java]: This function `getInstanceHelixDisabledType ` is added in change #2006 that enables batch. We could choose to remove the whole function?

Comment (qqu0127) [reply helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java]: removed the method and its use cases

File: helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java (hunk 2 above)
Comment (qqu0127) [reply helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java]: updated

File: helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java [hunk 4]
```diff
@@ -49,7 +43,7 @@ public void beforeClass() throws Exception {
     _accessor = new ConfigAccessor(_gZkClient);
   }
 
-  @Test
+  @Test(enabled = false)
```
Comment (desaikomal) [inline helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java:46]: May be it is just me, but i feel we should keep the test and make sure we get 'NotImplemented' exception. This way we know for sure that if anyone attempts to use batch APIs,  will get NotImplemented.

Comment (qqu0127) [reply helix-core/src/test/java/org/apache/helix/integration/TestBatchEnableInstances.java:52]: Good point. I found it hard to enable and modify existing tests, so I add a new test for this coverage. Thanks for the input.

**Review by desaikomal — APPROVED** [review]
thanks Quincy. Looks good to me.

**Comment by qqu0127** [comment]
This PR is ready to merge, approved by @desaikomal 
Commit message:
Revert and disable batch enable/disable functionality
Disable batch mode and remove related methods to fix backward compatibility issues introduced in 1.0.3

**Pull request merged**